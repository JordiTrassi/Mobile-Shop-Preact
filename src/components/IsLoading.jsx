import { CircularProgress, Typography } from '@mui/material';
import { Box } from '@mui/material';


export const IsLoading = () => {
  
  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        alignItems: 'center',
        mt: 20,
      }}>
      <Typography sx={{fontSize: 30}}>Is Loading ...</Typography>
      <CircularProgress sx={{mt: 3}} />
    </Box>
  );
}