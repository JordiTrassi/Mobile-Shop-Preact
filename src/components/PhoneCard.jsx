import { Link } from 'preact-router';
import { Box, Card, CardContent, CardMedia, IconButton, Tooltip, Typography } from '@mui/material';
import { AddCircle } from '@mui/icons-material';


export const PhoneCard = ({
    brand,
    imgUrl,
    model,
    price,
    id,
}) => {

  return (
      <Card
          className='animate__animated animate__fadeIn'
          sx={{ display: 'flex', mb: '20px', justifyContent: 'space-between', boxShadow: 6 }}
      >
      <Box sx={{ display: 'flex', flexDirection: 'column' }} >
        <CardContent sx={{ flex: '1 0 auto' }}>
          <Typography component="div" variant="h5" data-testid="test-collectionName">
            {brand}
          </Typography>
          <Typography variant="subtitle1" color="text.secondary" component="div">
            {model}
          </Typography>
        </CardContent>
              
        <Box sx={{ display: 'flex', alignItems: 'center', pl: 1, pb: 1 }}>
          <Link to={`/product/${id}`} >
            <Tooltip
              title="more info"
              placement="right"
              arrow
            >
              <IconButton>
                    <AddCircle fontSize='large'/> 
                </IconButton>              
            </Tooltip>
          </Link>
               
        </Box>
      </Box>
        <CardMedia
            component="img"
            image={imgUrl}
            alt={model}
            sx={{ height: 200, width: 'auto' }}     
        />
    </Card>
  );
}