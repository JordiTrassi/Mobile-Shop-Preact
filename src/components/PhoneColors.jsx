import { FormControl, FormControlLabel, FormLabel, Radio, RadioGroup } from '@mui/material';
import { useSelector } from 'react-redux';


export const PhoneColors = () => {
    const { colors } = useSelector(state => state.phoneList.selectedPhone);
    console.log(colors);

    return (
      
    <FormControl sx={{p: 2}}>
        <FormLabel id="demo-controlled-radio-buttons-group" sx={{ color: 'white', fontSize: 20}}>Select Color:</FormLabel>
        <RadioGroup
            // aria-labelledby="demo-controlled-radio-buttons-group"
            name="controlled-radio-buttons-group"
            // value={value}
            // onChange={handleChange}
          >
            {                   
                colors.map( color => (
                    <FormControlLabel value={color} control={<Radio />} label={color} sx={{ml: 2}} />    
                    )) 
            }

        </RadioGroup>
    </FormControl>
  )
}
