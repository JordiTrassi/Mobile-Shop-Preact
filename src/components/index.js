export * from './Header';
export * from './IsLoading';
export * from './PhoneCard';
export * from './PhoneColors';
export * from './PhoneInternalMemory';
export * from './ShopTimer';
