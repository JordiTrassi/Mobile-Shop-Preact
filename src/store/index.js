export * from './phoneListSlice';
export * from './store';
export * from './thunks';