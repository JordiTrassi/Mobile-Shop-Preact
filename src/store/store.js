import { configureStore } from '@reduxjs/toolkit';
import { phoneListSlice } from './phoneListSlice';

export const store = configureStore({
    reducer: {
        phoneList: phoneListSlice.reducer,
    },
});