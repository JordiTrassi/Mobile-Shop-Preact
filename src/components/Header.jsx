import { useState, useEffect } from 'preact/hooks';
import { useDispatch } from 'react-redux';
import { startLoadingPhones, getPhones } from '../store';
import { verifyInputValue } from '../helpers/verifyInputValue';

import { AppBar, Badge, Box, IconButton, InputBase, Toolbar, Tooltip, Typography, } from '@mui/material';
import { styled, alpha } from '@mui/material/styles';

import RingVolumeIcon from '@mui/icons-material/RingVolume';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import SearchIcon from '@mui/icons-material/Search';



const Search = styled('div')(({ theme }) => ({
  position: 'relative',
  borderRadius: theme.shape.borderRadius,
  backgroundColor: alpha(theme.palette.common.white, 0.15),
  '&:hover': {
    backgroundColor: alpha(theme.palette.common.white, 0.25),
  },
  marginRight: theme.spacing(2),
  marginLeft: 0,
  width: '100%',
  [theme.breakpoints.up('sm')]: {
    marginLeft: theme.spacing(3),
    width: 'auto',
  },
}));

const SearchIconWrapper = styled('div')(({ theme }) => ({
  padding: theme.spacing(0, 2),
  height: '100%',
  position: 'absolute',
  pointerEvents: 'none',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
}));

const StyledInputBase = styled(InputBase)(({ theme }) => ({
  color: 'inherit',
  '& .MuiInputBase-input': {
    padding: theme.spacing(1, 1, 1, 0),
    // vertical padding + font size from searchIcon
    paddingLeft: `calc(1em + ${theme.spacing(4)})`,
    transition: theme.transitions.create('width'),
    width: '100%',
    [theme.breakpoints.up('md')]: {
      width: '20ch',
    },
  },
}));

export const Header = () => {

  const dispatch = useDispatch();
  const [inputValue, setInputValue] = useState('');
  
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = async () => {
    const verifiedInputValue = await verifyInputValue(inputValue);
    dispatch(startLoadingPhones({ verifiedInputValue }));
    dispatch(getPhones({ verifiedInputValue }));
  };

  useEffect(() => {
    const listener = event => {
      if (event.code === "Enter" || event.code === "NumpadEnter") {
        event.preventDefault();
        onSubmit();
      }
    };
    document.addEventListener("keydown", listener);
    return () => {
      document.removeEventListener("keydown", listener);
    };
  }, [inputValue]);

  
  return (

    <Box sx={{ flexGrow: 1}}>
      <AppBar position="static" sx={{ borderRadius: 3, backgroundColor: '#4D4D4D' }}>
        <Toolbar>
          <Tooltip
            title="home"
            arrow
            placement="bottom"
          >
            <IconButton
              size="large"
              edge="start"
              color="inherit"
              aria-label="open drawer"
              sx={{ mr: 1 }}
              href="/"
            >
              <RingVolumeIcon />
            </IconButton>
          </Tooltip>    
          <Typography
            variant="h5"
            noWrap
            component="div"
            sx={{ display: { xs: 'none', sm: 'block' } }}
          >
            Mobile Shop
          </Typography>
          <Box sx={{ flexGrow: 1 }} />
          <Search>
            <SearchIconWrapper>
              <SearchIcon />
            </SearchIconWrapper>
            <StyledInputBase
              placeholder="Search mobile..."
              value={inputValue}
              onSubmit={onSubmit}
              onChange={onInputChange}
            />
          </Search>
          <Box sx={{ display: { xs: 'flex', md: 'flex' } }}>
            <Tooltip
            title="Your Cart"
            arrow
            placement="bottom"
            >
              <IconButton
                size="large"
                color="inherit"
              >
                <Badge badgeContent={1} color="error">
                  <ShoppingCartIcon />
                </Badge>
              </IconButton> 
            </Tooltip>
          </Box> 
        </Toolbar>
      </AppBar>
      
    </Box>
  );
}
