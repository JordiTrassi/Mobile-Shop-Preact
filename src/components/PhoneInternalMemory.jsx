import { useEffect, useState } from 'preact/hooks';
import { useDispatch, useSelector } from 'react-redux';
import { addSelectedPhoneMemory } from '../store/phoneListSlice';
import { FormControl, FormControlLabel, FormLabel, Radio, RadioGroup } from '@mui/material';


export const PhoneInternalMemory = () => {

  const dispatch = useDispatch();
  const { internalMemory } = useSelector(state => state.phoneList.selectedPhone);
  const [memory, setMemory] = useState(internalMemory[0]);
  
  const onSelectedMemory = ({target}) => {
        setMemory(target.value);
        dispatch(addSelectedPhoneMemory(target.value))
  }

  useEffect(() => {
    if (internalMemory.length === 1) {
        dispatch(addSelectedPhoneMemory(memory));
    }
  },[])


  return (
      
    <FormControl sx={{p: 2}}>
        <FormLabel id="demo-controlled-radio-buttons-group" sx={{ color: 'white', fontSize: 20}}>Select Memory:</FormLabel>
        <RadioGroup
            aria-labelledby="demo-controlled-radio-buttons-group"
            name="controlled-radio-buttons-group"
            value={memory}
            onChange={onSelectedMemory}
          >
            {                   
                internalMemory.map( memory => (
                    <FormControlLabel value={memory} control={<Radio />} label={memory} sx={{ml: 2}} />    
                    )) 
            }

        </RadioGroup>
    </FormControl>
  )
}