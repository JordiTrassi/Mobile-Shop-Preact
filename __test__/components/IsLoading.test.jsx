import { render, screen } from '@testing-library/preact';
import { h } from 'preact';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { IsLoading } from "../../src/components/IsLoading";
import { store } from '../../src/store/store'

describe('Pruebas en el componente <IsLoading />', () => {

    test('Debe de hacer match con el snapshot', () => {
        
        const { container } = render(
            <BrowserRouter>
                <Provider store={store}>
                    <IsLoading />
                </Provider>
            </BrowserRouter> 
        );

        expect(container).toMatchSnapshot();
    });

    
})
