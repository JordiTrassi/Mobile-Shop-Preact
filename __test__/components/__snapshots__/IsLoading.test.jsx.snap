// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente <IsLoading /> Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-cn3mpb"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-f8nw7v-MuiTypography-root"
    >
      Is Loading ...
    </p>
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1n51an3-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</div>
`;
